automation/taskcluster/decision_task_snapshot_release.py

```diff
@@ -0,0 +1,184 @@
+                 "-Pcoverage -Psnapshot " +
+                 " ".join(["{}:{}".format(module_name, gradle_task) for gradle_task in ['assemble', 'test', 'lint']])  +
+                 " uploadArchives zipMavenArtifacts " +
+                 " && automation/taskcluster/action/upload_coverage_report.sh"),
+        features={
+            "chainOfTrust": True,
+            "taskclusterProxy": True
+        },
+        worker_type='gecko-focus',
+        scopes = [
+            "secrets:get:project/mobile/android-components/public-tokens"
+        ],
+        artifacts=artifacts,
+    )
+
+
+def generate_beetmover_task(build_task_id, info, version):
+    module_name = ":{}".format(info['name'])
+    upstream_artifacts = [
+        {
+            "paths": [
+                info['artifact'],
+            ],
+            "taskId": build_task_id,
+            "taskType": "build",
+            "zipExtract": True
+        }
+    ]
+    # TODO: switch to prod bucket once staging works smoothly
+    scopes = [
+        "project:mobile:android-components:releng:beetmover:bucket:maven-snapshot-staging",
+        "project:mobile:android-components:releng:beetmover:action:push-to-maven"
+    ]
+    # TODO: switch to prod mobile-beetmoverwoerk-v1
+    return taskcluster.slugId(), BUILDER.beetmover_task(
+        name="Android Components - Publish Snapshot Module {} via beetmover".format(module_name),
+        description="Publish snapshot release module {} to https://maven.mozilla.org".format(module_name),
+        version=version,
+        artifact_id=info['name'],
+        dependencies=[build_task_id],
+        upstreamArtifacts=upstream_artifacts,
+        scopes=scopes,
+        worker_type='mobile-beetmover-dev',
+        is_snapshot=True
+    )
+
+
+def generate_raw_task(name, description, command_to_run):
+    checkout = ("export TERM=dumb && git fetch {} {} && "
+                "git config advice.detachedHead false && "
+                "git checkout {} && ".format(REPO_URL, BRANCH, HEAD_REV))
+    return taskcluster.slugId(), BUILDER.raw_task(
+        name=name,
+        description=description,
+        command=(checkout +
+                 command_to_run)
+    )
+
+
+def generate_detekt_task():
+    return generate_raw_task(
+        name='Android Components - detekt',
+        description='Running detekt over all modules',
+        command_to_run='./gradlew --no-daemon clean detekt')
+
+
+def generate_ktlint_task():
+    return generate_raw_task(
+        name='Android Components - ktlint',
+        description='Running ktlint over all modules',
+        command_to_run='./gradlew --no-daemon clean ktlint')
+
+
+def generate_compare_locales_task():
+    return generate_raw_task(
+        name='Android Components - compare-locales',
+        description='Validate strings.xml with compare-locales',
+        command_to_run='pip install "compare-locales>=4.0.1,<5.0" && compare-locales --validate l10n.toml .')
```

Comment:
There's quite some duplication here in regards to how we setup our build pipeline when looking at the other decision task scripts (e.g. run detekt, ktlint, compare-locales, ..). I wonder if we can somehow make that easier to maintain in a single place in the future?